### PR TITLE
[Feature] [SR SQL Planner Summer Camp] add dropRepository analyzer

### PR DESCRIPTION
**Pull request by melt-code**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7951

## Problem Summary(Required) ：

add dropRepository analyzer

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 26 (80.77%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[418]|
|:large_blue_circle:|com/starrocks/analysis/DropRepositoryStmt.java|1|2|50.00%|[72]|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropRepositoryAnalyzer.java|14|17|82.35%|[16, 32, 38]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 23 / 29 (79.31%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[443]|
|:large_blue_circle:|com/starrocks/analysis/DropRepositoryStmt.java|1|2|50.00%|[72]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[943]|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropRepositoryAnalyzer.java|14|17|82.35%|[16, 32, 44]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropRepositoryAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+public class DropRepositoryAnalyzer {
+
+    public static void analyze(DropRepositoryStmt dropRepositoryStmt, ConnectContext session) {
+        new DropRepositoryAnalyzerVisitor().analyze(dropRepositoryStmt, session);
+    }
+
+    public static class DropRepositoryAnalyzerVisitor extends AstVisitor<Void, ConnectContext> {
+        public void analyze(DdlStmt statement, ConnectContext session) {
+            visit(statement, session);
+
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropRepositoryAnalyzer.java:25]:
```suggestion
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 20 / 26 (76.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[422]|
|:large_blue_circle:|com/starrocks/analysis/DropRepositoryStmt.java|1|2|50.00%|[72]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[953]|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropRepositoryAnalyzer.java|14|17|82.35%|[16, 31, 43]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java [hunk 2]
```diff
@@ -438,6 +439,10 @@ public R visitShowHistogramStatsMetaStatement(ShowHistogramStatsMetaStmt stateme
         return visitShowStatement(statement, context);
     }
 
+    public R visitDropRepositoryStmt(DropRepositoryStmt statement, C context) {
```
[the remark this answers is not in the document]
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java]: It has been modified

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 3]
```diff
@@ -2380,6 +2381,12 @@ public ParseNode visitShowProcesslistStatement(StarRocksParser.ShowProcesslistSt
         return new ShowProcesslistStmt(isShowFull);
     }
 
+    // ------------------------------------------- Backup Store Statement ----------------------------------------------
```
[the remark this answers is not in the document]
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java]: It has been modified

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropRepositoryAnalyzer.java (hunk 1 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropRepositoryAnalyzer.java]: It has been modified

**Comment by melt-code** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 23 / 28 (82.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[458]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[1278]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateDropRepositoryAnalyzer.java|15|18|83.33%|[18, 73, 82]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DropRepositoryStmt.java|2|2|100.00%|[]|

**Comment by melt-code** [comment]
run starrocks_admit_test

**Review by HangyuanLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateDropRepositoryAnalyzer.java [hunk 4]
```diff
@@ -4,31 +4,36 @@
...
 import com.starrocks.common.AnalysisException;
 import com.starrocks.common.ErrorCode;
 import com.starrocks.common.ErrorReport;
 import com.starrocks.common.FeNameFormat;
 import com.starrocks.qe.ConnectContext;
+import com.starrocks.server.GlobalStateMgr;
 import com.starrocks.sql.ast.AstVisitor;
 
-public class CreateRepositoryAnalyzer {
+public class CreateDropRepositoryAnalyzer {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateDropRepositoryAnalyzer.java:18]:
```suggestion
public class RepositoryAnalyzer {
```

**Comment by melt-code** [comment]
run starrocks_admit_test

**Comment by melt-code** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 23 / 28 (82.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[460]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|2|3|66.67%|[1278]|
|:large_blue_circle:|com/starrocks/sql/analyzer/RepositoryAnalyzer.java|15|18|83.33%|[18, 73, 82]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/DropRepositoryStmt.java|2|2|100.00%|[]|

**Comment by melt-code** [comment]
run starrocks_admit_test

**Review by HangyuanLiu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**